### ISSUE TITLE
iasl: Unable to re-build TPM2.dsl without optional trailing options

### ISSUE THREAD

**Issue by gemarcano** [acpica/acpica#984]
This is apparently a known issue in the source code, but I don't see any open issues about it. Here is the relevant comment in the source code for iasl:
https://github.com/acpica/acpica/blob/437f7e1a0039d227d3952677c3f9f5ad9c814395/source/compiler/dttable2.c#L2925

I'm (once again) rebuilding the ACPI tables for my laptop, and its TPM2 table doesn't use any of the optional trailing fields, so even if I wanted to fill in fake data to make iasl happy, I'm not sure I can just make up information for the log size and log location if these don't really exist.

**Comment by SaketADumbre** [comment]
I'll need to look into this more in detail! From the top, I will need some clarifications regarding if those optional fields are even really optional, since the TPMA table in the same TCG document lists those 2 fields as required (not optional) but in the meantime,  I would really appreciate any more zoomed in details like the debug log showing more precise point of failure. The reason for that is my predecessor was in charge of the said comment/commit and I am not sure exactly what exactly his TBD was and how much did he manage to get in a few years ago! Thanks a lot!

**Comment by SaketADumbre** [comment]
Hi @gemarcano can you also please share your test tpm2 table or better yet the entire acpidump from the affected system? Thanks!

**Comment by gemarcano** [comment]
I know the ACPI TPM2 table is bad to begin with on my system (missing the start method argument I think?), but I tried rebuilding the TPM2 table after adding 4 bytes as required for start method 2. I'm attaching a complete ACPI dump to this comment (for reference, my laptop is a Dell G5 5505 SE on BIOS version 1.24).

From what I can tell from the `TCG ACPI Specification | Version 1.4 | Revision 15 | 4/3/2024`, for TPM 2.0 the LAML and LASA fields are optional. Interestingly enough for TPM1.2 they seem mandatory.

[dell_G5_5505_SE_BIOS_1_24_acpi.bin.zip](https://github.com/user-attachments/files/18069062/dell_G5_5505_SE_BIOS_1_24_acpi.bin.zip)

**Comment by SaketADumbre** [comment]
You are right. I doubt the TPM2 implementation is updated to 2.0 in our codebase as it is also missing the AMD mailbox usage and a few more fields. Thanks for the data, I will follow up on this after some more analysis and deeper understanding of what is going on!

**Comment by gemarcano** [comment]
Ok, here is the true ACPI table dump for my Dell G5 5505 SE laptop with BIOS version 1.24. The previous file had a bunch of fixes I had made to a bunch of tables so it isn't representative of what Dell pushed out the door. This one is.

[acpi_dump_dell_g5_5505_se_bios_1_24.bin.zip](https://github.com/user-attachments/files/18082730/acpi_dump_dell_g5_5505_se_bios_1_24.bin.zip)
